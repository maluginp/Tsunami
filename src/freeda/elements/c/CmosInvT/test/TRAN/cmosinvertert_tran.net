* Thermal CMOS Inverter DC Analysis

**** This netlist is for DC Analysis of Thermal Cmos Inverter. The Inverter is based on EKV mosfet model
.options initTmp=300
.ref "tref"
.ref 0

.tran2 tstop=4e-6 tstep=1e-10
vpulse:Vin2 22 0 v1=0 v2=5 per=2e-6 pw=1e-6 tr=0.002e-6 tf=0.002e-6

*** Vdd connection 
vsource:Vdd 1 0 vdc=5

***** Instantiation of Thermal Cmos Inverter
cmosinvt:invt 1 2 3 0 2000 "tref" tmp=initTmp wn=20e-6 wp=50e-6

* Source Resistance
r:Rsource 22 2 r=10
* Source Capacitance
c:Csource 2 0 c=1e-12

**** Load Resistance
r:Rout 3 0 r=1000000
**** Load Capacitance
c:Cout 3 0 c=1e-12

****Thermal 
r:Rthermal 2000 2001 r=20e3
c:Cthermal 2000 2001 c=1e-12
vsource:vt1 2001 "tref" vdc=initTmp

***Plot
.options gnuplot
.out plot term 3 vt in "Vout_300K_20k.out"
.out plot term 2000 vt in "JunctionTemp_300K_20k.out"
.out plot element "vsource:vt1" 0 it in "Iheat_300K_20k.out"
**.out plot element "cmosinvt:invt" 0 it in "inv_current.out"
**.out plot term 2 vt in "vin.out"
.end
