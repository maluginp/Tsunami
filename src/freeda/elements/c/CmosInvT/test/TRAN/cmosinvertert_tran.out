
**********  fREEDA 2.0 running on Tue Jul  7 17:40:36 2009  **********

**  Environment variables: **
FREEDA_HOME = /spriyad/freeda
FREEDA_LIBRARY = /spriyad/freeda/library
FREEDA_PROJECTS = /spriyad/freeda/projects
FREEDA_PATH = /spriyad/freeda/freeda
FREEDA_BIN = /spriyad/freeda/freeda/bin
FREEDA_SIMULATOR = /spriyad/freeda/freeda/simulator
FREEDA_ELEMENTS = /spriyad/freeda/freeda/simulator/elements
FREEDA_DOCUMENTATION = /tmp
FREEDA_WEB_DOCUMENTATION = http://www.freeda.org/doc
FREEDA_BROWSER = firefox

* Thermal CMOS Inverter DC Analysis

* Thermal CMOS Inverter DC Analysis

**** This netlist is for DC Analysis of Thermal Cmos Inverter. The Inverter is based on EKV mosfet model
.options inittmp=300
.ref "tref"
.ref 0

.tran2 tstop=4e-6 tstep=1e-10
vpulse:vin2 22 0 v1=0 v2=5 per=2e-6 pw=1e-6 tr=0.002e-6 tf=0.002e-6

*** Vdd connection 
vsource:vdd 1 0 vdc=5

***** Instantiation of Thermal Cmos Inverter
cmosinvt:invt 1 2 3 0 2000 "tref" tmp=inittmp wn=20e-6 wp=50e-6

* Source Resistance
r:rsource 22 2 r=10
* Source Capacitance
c:csource 2 0 c=1e-12

**** Load Resistance
r:rout 3 0 r=1000000
**** Load Capacitance
c:cout 3 0 c=1e-12

****Thermal 
r:rthermal 2000 2001 r=20e3
c:cthermal 2000 2001 c=1e-12
vsource:vt1 2001 "tref" vdc=inittmp

***Plot
.options gnuplot
.out plot term 3 vt in "vout_300k_20k.out"
.out plot term 2000 vt in "junctiontemp_300k_20k.out"
.out plot element "vsource:vt1" 0 it in "iheat_300k_20k.out"
**.out plot element "cmosinvt:invt" 0 it in "inv_current.out"
**.out plot term 2 vt in "vin.out"

   *** Starting analysis ...

-----------------------------------------------------------------------------
   --- Starting transient simulation ...

-----------------------------------------------------------------------------
   |   Step   |    Time (s)    |   Residual (V)   |
-----------------------------------------------------------------------------
   Using Line Search Method - (Back Track).
   |     200  |  2.000000e-08  |  9.664278e-12    |
   |     400  |  4.000000e-08  |  1.969591e-03    |
   |     600  |  6.000000e-08  |  1.421220e-12    |
   |     800  |  8.000000e-08  |  5.174584e-04    |
   |    1000  |  1.000000e-07  |  5.863109e-04    |
   |    1200  |  1.200000e-07  |  7.170194e-04    |
   |    1400  |  1.400000e-07  |  2.453347e-04    |
   |    1600  |  1.600000e-07  |  1.838927e-03    |
   |    1800  |  1.800000e-07  |  1.436694e-03    |
   |    2000  |  2.000000e-07  |  9.967634e-04    |
   |    2200  |  2.200000e-07  |  5.202340e-04    |
   |    2400  |  2.400000e-07  |  4.018186e-07    |
   |    2600  |  2.600000e-07  |  1.775616e-03    |
   |    2800  |  2.800000e-07  |  1.321466e-03    |
   |    3000  |  3.000000e-07  |  8.696184e-04    |
   |    3200  |  3.200000e-07  |  4.536189e-04    |
   |    3400  |  3.400000e-07  |  1.909712e-03    |
   |    3600  |  3.600000e-07  |  1.547081e-03    |
   |    3800  |  3.800000e-07  |  1.117095e-03    |
   |    4000  |  4.000000e-07  |  5.814169e-04    |
   |    4200  |  4.200000e-07  |  2.393116e-04    |
   |    4400  |  4.400000e-07  |  1.986911e-03    |
   |    4600  |  4.600000e-07  |  1.401030e-03    |
   |    4800  |  4.800000e-07  |  9.732014e-04    |
   |    5000  |  5.000000e-07  |  5.063094e-04    |
   |    5200  |  5.200000e-07  |  3.811568e-07    |
   |    5400  |  5.400000e-07  |  1.733316e-03    |
   |    5600  |  5.600000e-07  |  1.288682e-03    |
   |    5800  |  5.800000e-07  |  8.058283e-04    |
   |    6000  |  6.000000e-07  |  4.417157e-04    |
   |    6200  |  6.200000e-07  |  3.678381e-07    |
   |    6400  |  6.400000e-07  |  1.509307e-03    |
   |    6600  |  6.600000e-07  |  1.122807e-03    |
   |    6800  |  6.800000e-07  |  5.677387e-04    |
   |    7000  |  7.000000e-07  |  4.023418e-07    |
   |    7200  |  7.200000e-07  |  1.936132e-03    |
   |    7400  |  7.400000e-07  |  1.442210e-03    |
   |    7600  |  7.600000e-07  |  9.490067e-04    |
   |    7800  |  7.800000e-07  |  3.554603e-07    |
   |    8000  |  8.000000e-07  |  1.671539e-03    |
   |    8200  |  8.200000e-07  |  1.242843e-03    |
   |    8400  |  8.400000e-07  |  7.781926e-04    |
   |    8600  |  8.600000e-07  |  2.700921e-04    |
   |    8800  |  8.800000e-07  |  3.421201e-07    |
   |    9000  |  9.000000e-07  |  1.581667e-03    |
   |    9200  |  9.200000e-07  |  1.084319e-03    |
   |    9400  |  9.400000e-07  |  6.785471e-04    |
   |    9600  |  9.600000e-07  |  3.742046e-07    |
   |    9800  |  9.800000e-07  |  1.763668e-03    |
   |   10000  |  1.000000e-06  |  1.392742e-03    |
   |   10200  |  1.020000e-06  |  6.365291e-12    |
   |   10400  |  1.040000e-06  |  8.493253e-13    |
   |   10600  |  1.060000e-06  |  1.135506e-13    |
   |   10800  |  1.080000e-06  |  1.110924e-03    |
   |   11000  |  1.100000e-06  |  1.643135e-03    |
   |   11200  |  1.120000e-06  |  1.544935e-03    |
   |   11400  |  1.140000e-06  |  1.520099e-03    |
   |   11600  |  1.160000e-06  |  4.265875e-07    |
   |   11800  |  1.180000e-06  |  6.189434e-04    |
   |   12000  |  1.200000e-06  |  1.249856e-03    |
   |   12200  |  1.220000e-06  |  1.890602e-03    |
   |   12400  |  1.240000e-06  |  3.261466e-04    |
   |   12600  |  1.260000e-06  |  9.855794e-04    |
   |   12800  |  1.280000e-06  |  1.547839e-03    |
   |   13000  |  1.300000e-06  |  4.961153e-07    |
   |   13200  |  1.320000e-06  |  8.764246e-04    |
   |   13400  |  1.340000e-06  |  1.473272e-03    |
   |   13600  |  1.360000e-06  |  4.191264e-07    |
   |   13800  |  1.380000e-06  |  6.145725e-04    |
   |   14000  |  1.400000e-06  |  1.239224e-03    |
   |   14200  |  1.420000e-06  |  1.877387e-03    |
   |   14400  |  1.440000e-06  |  3.230303e-04    |
   |   14600  |  1.460000e-06  |  9.776456e-04    |
   |   14800  |  1.480000e-06  |  1.537962e-03    |
   |   15000  |  1.500000e-06  |  2.650290e-04    |
   |   15200  |  1.520000e-06  |  8.706315e-04    |
   |   15400  |  1.540000e-06  |  1.463143e-03    |
   |   15600  |  1.560000e-06  |  4.134211e-07    |
   |   15800  |  1.580000e-06  |  6.094412e-04    |
   |   16000  |  1.600000e-06  |  1.229526e-03    |
   |   16200  |  1.620000e-06  |  1.860135e-03    |
   |   16400  |  1.640000e-06  |  3.208992e-04    |
   |   16600  |  1.660000e-06  |  9.697543e-04    |
   |   16800  |  1.680000e-06  |  1.629757e-03    |
   |   17000  |  1.700000e-06  |  2.623560e-04    |
   |   17200  |  1.720000e-06  |  7.379693e-04    |
   |   17400  |  1.740000e-06  |  1.451888e-03    |
   |   17600  |  1.760000e-06  |  4.070463e-07    |
   |   17800  |  1.780000e-06  |  6.056431e-04    |
   |   18000  |  1.800000e-06  |  1.221223e-03    |
   |   18200  |  1.820000e-06  |  1.847250e-03    |
   |   18400  |  1.840000e-06  |  3.178524e-04    |
   |   18600  |  1.860000e-06  |  9.619656e-04    |
   |   18800  |  1.880000e-06  |  1.617089e-03    |
   |   19000  |  1.900000e-06  |  4.011914e-07    |
   |   19200  |  1.920000e-06  |  7.326153e-04    |
   |   19400  |  1.940000e-06  |  1.441899e-03    |
   |   19600  |  1.960000e-06  |  4.015020e-07    |
   |   19800  |  1.980000e-06  |  6.005971e-04    |
   |   20000  |  2.000000e-06  |  1.211676e-03    |
   |   20200  |  2.020000e-06  |  2.126147e-11    |
   |   20400  |  2.040000e-06  |  2.785576e-12    |
   |   20600  |  2.060000e-06  |  1.421220e-12    |
   |   20800  |  2.080000e-06  |  1.141683e-03    |
   |   21000  |  2.100000e-06  |  7.073535e-04    |
   |   21200  |  2.120000e-06  |  2.443476e-04    |
   |   21400  |  2.140000e-06  |  1.986152e-03    |
   |   21600  |  2.160000e-06  |  1.397069e-03    |
   |   21800  |  2.180000e-06  |  9.696931e-04    |
   |   22000  |  2.200000e-06  |  5.061003e-04    |
   |   22200  |  2.220000e-06  |  3.791886e-07    |
   |   22400  |  2.240000e-06  |  1.725010e-03    |
   |   22600  |  2.260000e-06  |  1.285517e-03    |
   |   22800  |  2.280000e-06  |  8.037568e-04    |
   |   23000  |  2.300000e-06  |  4.406748e-04    |
   |   23200  |  2.320000e-06  |  3.651491e-07    |
   |   23400  |  2.340000e-06  |  1.504998e-03    |
   |   23600  |  2.360000e-06  |  1.120887e-03    |
   |   23800  |  2.380000e-06  |  5.648473e-04    |
   |   24000  |  2.400000e-06  |  3.989849e-07    |
   |   24200  |  2.420000e-06  |  1.932835e-03    |
   |   24400  |  2.440000e-06  |  1.437658e-03    |
   |   24600  |  2.460000e-06  |  9.454469e-04    |
   |   24800  |  2.480000e-06  |  4.925008e-04    |
   |   25000  |  2.500000e-06  |  3.606966e-07    |
   |   25200  |  2.520000e-06  |  1.683802e-03    |
   |   25400  |  2.540000e-06  |  1.251949e-03    |
   |   25600  |  2.560000e-06  |  7.839022e-04    |
   |   25800  |  2.580000e-06  |  2.720717e-04    |
   |   26000  |  2.600000e-06  |  3.471541e-07    |
   |   26200  |  2.620000e-06  |  1.593260e-03    |
   |   26400  |  2.640000e-06  |  6.753700e-04    |
   |   26600  |  2.660000e-06  |  3.720960e-07    |
   |   26800  |  2.680000e-06  |  1.760313e-03    |
   |   27000  |  2.700000e-06  |  1.386508e-03    |
   |   27200  |  2.720000e-06  |  8.677516e-04    |
   |   27400  |  2.740000e-06  |  4.756180e-04    |
   |   27600  |  2.760000e-06  |  4.257814e-07    |
   |   27800  |  2.780000e-06  |  1.623882e-03    |
   |   28000  |  2.800000e-06  |  1.209038e-03    |
   |   28200  |  2.820000e-06  |  7.570038e-04    |
   |   28400  |  2.840000e-06  |  2.623962e-04    |
   |   28600  |  2.860000e-06  |  1.966500e-03    |
   |   28800  |  2.880000e-06  |  1.538637e-03    |
   |   29000  |  2.900000e-06  |  1.069034e-03    |
   |   29200  |  2.920000e-06  |  6.591866e-04    |
   |   29400  |  2.940000e-06  |  2.286021e-04    |
   |   29600  |  2.960000e-06  |  1.715694e-03    |
   |   29800  |  2.980000e-06  |  1.341592e-03    |
   |   30000  |  3.000000e-06  |  8.452416e-04    |
   |   30200  |  3.020000e-06  |  6.366254e-12    |
   |   30400  |  3.040000e-06  |  8.482723e-13    |
   |   30600  |  3.060000e-06  |  1.135487e-13    |
   |   30800  |  3.080000e-06  |  1.111013e-03    |
   |   31000  |  3.100000e-06  |  1.643267e-03    |
   |   31200  |  3.120000e-06  |  1.545060e-03    |
   |   31400  |  3.140000e-06  |  1.235212e-03    |
   |   31600  |  3.160000e-06  |  1.763647e-03    |
   |   31800  |  3.180000e-06  |  3.028138e-04    |
   |   32000  |  3.200000e-06  |  9.158399e-04    |
   |   32200  |  3.220000e-06  |  1.539301e-03    |
   |   32400  |  3.240000e-06  |  4.574991e-07    |
   |   32600  |  3.260000e-06  |  6.420986e-04    |
   |   32800  |  3.280000e-06  |  1.294705e-03    |
   |   33000  |  3.300000e-06  |  1.782111e-03    |
   |   33200  |  3.320000e-06  |  5.255770e-04    |
   |   33400  |  3.340000e-06  |  1.151530e-03    |
   |   33600  |  3.360000e-06  |  1.742126e-03    |
   |   33800  |  3.380000e-06  |  3.009828e-04    |
   |   34000  |  3.400000e-06  |  9.096084e-04    |
   |   34200  |  3.420000e-06  |  1.528637e-03    |
   |   34400  |  3.440000e-06  |  4.512621e-07    |
   |   34600  |  3.460000e-06  |  6.357151e-04    |
   |   34800  |  3.480000e-06  |  1.282547e-03    |
   |   35000  |  3.500000e-06  |  1.940336e-03    |
   |   35200  |  3.520000e-06  |  3.351786e-04    |
   |   35400  |  3.540000e-06  |  1.100305e-03    |
   |   35600  |  3.560000e-06  |  1.730018e-03    |
   |   35800  |  3.580000e-06  |  2.977048e-04    |
   |   36000  |  3.600000e-06  |  9.023452e-04    |
   |   36200  |  3.620000e-06  |  1.516850e-03    |
   |   36400  |  3.640000e-06  |  4.442855e-07    |
   |   36600  |  3.660000e-06  |  6.327624e-04    |
   |   36800  |  3.680000e-06  |  1.273915e-03    |
   |   37000  |  3.700000e-06  |  1.926949e-03    |
   |   37200  |  3.720000e-06  |  3.315633e-04    |
   |   37400  |  3.740000e-06  |  1.045004e-03    |
   |   37600  |  3.760000e-06  |  1.716830e-03    |
   |   37800  |  3.780000e-06  |  2.961474e-04    |
   |   38000  |  3.800000e-06  |  8.950074e-04    |
   |   38200  |  3.820000e-06  |  1.504107e-03    |
   |   38400  |  3.840000e-06  |  4.368947e-07    |
   |   38600  |  3.860000e-06  |  6.264894e-04    |
   |   38800  |  3.880000e-06  |  1.263926e-03    |
   |   39000  |  3.900000e-06  |  1.912184e-03    |
   |   39200  |  3.920000e-06  |  3.298880e-04    |
   |   39400  |  3.940000e-06  |  1.036666e-03    |
   |   39600  |  3.960000e-06  |  1.704891e-03    |
   |   39800  |  3.980000e-06  |  2.929322e-04    |
   |   40000  |  4.000000e-06  |  8.864943e-04    |
-----------------------------------------------------------------------------

   --- Residual: 0.000886494
   --- Writing output vectors ...
 Plotting output file: vout_300k_20k.out.
 Plotting output file: junctiontemp_300k_20k.out.
 Plotting output file: iheat_300k_20k.out.


**************  fREEDA 2.0 stopping on Tue Jul  7 17:40:42 2009  ***********
